deepmd/pt/utils/stat.py

```
@@ -35,51 +32,179 @@
+            sys = datasets[sys_index]
+            for frame in frames:
+                newele_counter += 1
+                if not newele_counter > min_frames_per_element_forstat:
+                    frame_data = sys.__getitem__(frame)
+                    sys_stat_new = {}
+                    for dd in frame_data:
+                        if dd == "type":
+                            continue
+                        if frame_data[dd] is None:
+                            sys_stat_new[dd] = None
+                        elif isinstance(frame_data[dd], np.ndarray):
```

Ruff (0.8.2)

Undefined name `np`

(F821)